config/config.go

```diff
@@ -1053,3 +1056,111 @@ func (m Matchers) MarshalJSON() ([]byte, error) {
+
+	GzipCompression = "gzip"
+)
+
+// UnmarshalYAML implements the yaml.Unmarshaler interface.
+func (t *TracingClientType) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	*t = TracingClientType("")
+	type plain TracingClientType
+	if err := unmarshal((*plain)(t)); err != nil {
+		return err
+	}
+
+	if *t != TracingClientHTTP && *t != TracingClientGRPC {
+		return fmt.Errorf("expected tracing client type to be to be %s or %s, but got %s",
+			TracingClientHTTP, TracingClientGRPC, *t,
+		)
+	}
+
+	return nil
+}
+
+// TracingConfig configures the tracing options.
+type TracingConfig struct {
+	ClientType       TracingClientType `yaml:"client_type,omitempty"`
+	Endpoint         string            `yaml:"endpoint,omitempty"`
+	SamplingFraction float64           `yaml:"sampling_fraction,omitempty"`
+	Insecure         bool              `yaml:"insecure,omitempty"`
+	TLSConfig        config.TLSConfig  `yaml:"tls_config,omitempty"`
+	Headers          map[string]string `yaml:"headers,omitempty"`
+	Compression      string            `yaml:"compression,omitempty"`
+	Timeout          model.Duration    `yaml:"timeout,omitempty"`
+}
+
+// SetDirectory joins any relative file paths with dir.
+func (t *TracingConfig) SetDirectory(dir string) {
+	t.TLSConfig.SetDirectory(dir)
+}
+
+// UnmarshalYAML implements the yaml.Unmarshaler interface.
+func (t *TracingConfig) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	*t = TracingConfig{
+		ClientType: TracingClientGRPC,
+	}
+	type plain TracingConfig
+	if err := unmarshal((*plain)(t)); err != nil {
+		return err
+	}
+
+	if err := validateHeadersForTracing(t.Headers); err != nil {
+		return err
+	}
+
+	if t.Endpoint == "" {
+		return errors.New("tracing endpoint must be set")
+	}
+
+	if t.Compression != "" && t.Compression != GzipCompression {
+		return fmt.Errorf("invalid compression type %s provided, valid options: %s",
+			t.Compression, GzipCompression)
+	}
+
+	return nil
+}
+
+var reservedHeaders = map[string]struct{}{
+	// NOTE: authorization is checked specially,
+	// see RemoteWriteConfig.UnmarshalYAML.
+	// "authorization":                  {},
+	"host":                              {},
+	"content-encoding":                  {},
+	"content-length":                    {},
+	"content-type":                      {},
+	"user-agent":                        {},
+	"connection":                        {},
+	"keep-alive":                        {},
+	"proxy-authenticate":                {},
+	"proxy-authorization":               {},
+	"www-authenticate":                  {},
+	"accept-encoding":                   {},
+	"x-prometheus-remote-write-version": {},
+	"x-prometheus-remote-read-version":  {},
+
+	// Added by SigV4.
+	"x-amz-date":           {},
+	"x-amz-security-token": {},
+	"x-amz-content-sha256": {},
+}
+
+func validateHeadersForTracing(headers map[string]string) error {
+	for header := range headers {
+		if strings.ToLower(header) == "authorization" {
```

Comment:
This is also copied from `prometheus/prometheus`: https://github.com/prometheus/prometheus/blob/main/config/config.go#L1083

The second quoted sentence refers to the field value, not the name. It's not permissible to have whitespace in names at all (they're just `token`s - i.e. one or more non-control/non-separator character). So even if someone had a configuration with a header that included whitespace, it would likely be ignored or discarded later on.

Reply:
Ah – I misread! 🙂